src/generate/azgenerator.ts

```diff
@@ -52,31 +52,30 @@ export async function processRequest(host: Host) {
         }
 
         for (const f in files) {
-            if (!isNullOrUndefined(files[f])) {
-                if (
-                    (AzConfiguration.getValue(CodeGenConstants.generationMode) !==
-                        GenerationMode.Incremental &&
-                        (f.endsWith('azext_metadata.json') ||
-                            (f.endsWith('setup.py') &&
-                                AzConfiguration.getValue(CodeGenConstants.targetMode) !==
-                                    TargetMode.Core))) ||
-                    f.endsWith('HISTORY.rst') ||
-                    f.endsWith('setup.cfg') ||
-                    // f.endsWith('report.md') ||
-                    f.endsWith('tests/__init__.py') ||
-                    f.endsWith('preparers.py')
-                ) {
-                    host.WriteFile(f, files[f]);
-                } else {
-                    host.WriteFile(f, files[f].join(EOL));
-                }
+            if (
+                (AzConfiguration.getValue(CodeGenConstants.generationMode) !==
+                    GenerationMode.Incremental &&
+                    (f.endsWith('azext_metadata.json') ||
+                        (f.endsWith('setup.py') &&
+                            AzConfiguration.getValue(CodeGenConstants.targetMode) !==
+                                TargetMode.Core))) ||
+                f.endsWith('HISTORY.rst') ||
```

Comment:
Can you put file name check into a separate function?

Reply:
Actually that's the logic to decide whether we go the old way or the new templates rendering way not for file name check,  because the old string concatenation is return a string array we need to join it as whole file content, but the template based way doesn;t need this. Nevertheless,  I just find a new way of using the return result type to decide.